### PR TITLE
Fix multiple issues with user population in tournament client

### PR DESCRIPTION
**Pull request by frenzibyte**
This fixes multiple issues with user population in tournament clients, spotted while I was looking at re-fetching user on null global rank (see https://github.com/ppy/osu/pull/11708#discussion_r576599990).

- https://github.com/ppy/osu/commit/3b4e02e5c78fd77325f188c0379785e47b61d6aa
   This fixes user population on bracket loading not being performed synchronously, opposed to `addBeatmaps`, as the API might be busy performing other requests while the user request is hanging there in the queue, causing the brackets to be saved with unpopulated user instances.

- https://github.com/ppy/osu/commit/85ebc8e06cd45be8e66f40c059dac6b20c926cc8
  This fixes user requests failure overwriting the user ID, which causes issues if the failed request was to be performed after the user ID is set to a valid one. For example, run a tournament client on `master` and do the following:
  1. go to team editor screen and add a player 
  2. set to an invalid ID then quickly to a valid ID before the user request with the invalid ID is performed
  3. wait until user requests are performed then save changes and re-run the tournament again.
  
  expected: the added player would have the valid ID saved to the bracket file from the previous session.
  actual: the added player [would have ID `1`](https://github.com/ppy/osu/blob/bef0e5cfa17c79f352528cd559ffa9a10b011d31/osu.Game.Tournament/TournamentGameBase.cs#L230) saved to the bracket file from the previous session, because of the failed request overwriting the valid user ID.

EDIT: Last two commits (https://github.com/ppy/osu/commit/705e9267497bc1eec01da9d73ec2630ad2be327d  and https://github.com/ppy/osu/commit/85ebc8e06cd45be8e66f40c059dac6b20c926cc8) are reverted with  https://github.com/ppy/osu/pull/11799/commits/61bf9a64bb117483093318b99ce135cd50a2e8c0 instead, now the user ID will become overwritten if the successful request isn't performed afterwards, which makes sense somewhat.

### CODE REVIEW

File: osu.Game.Tournament/TournamentGameBase.cs [hunk 1]
```diff
@@ -225,13 +225,12 @@ public void PopulateUser(User user, Action success = null, Action failure = null
                 success?.Invoke();
             };
 
-            req.Failure += _ =>
-            {
-                user.Id = 1;
```
Comment (peppy) [inline osu.Game.Tournament/TournamentGameBase.cs:230]: have you checked the blame for this? it looks to be intentional.

File: osu.Game.Tournament/Screens/Editors/TeamEditorScreen.cs [hunk 2]
```diff
@@ -277,7 +277,8 @@ private void load()
                         userId.Value = user.Id.ToString();
                         userId.BindValueChanged(idString =>
                         {
-                            int.TryParse(idString.NewValue, out var parsed);
+                            if (!(int.TryParse(idString.NewValue, out var parsed)))
```
Comment (peppy) [inline osu.Game.Tournament/Screens/Editors/TeamEditorScreen.cs:280]: won't this mean that a panel stays looking like it is a valid use even though the id has changed?

File: osu.Game.Tournament/TournamentGameBase.cs (hunk 1 above)
Comment (frenzibyte) [reply osu.Game.Tournament/TournamentGameBase.cs:230]: Seems to have came from this commit https://github.com/ppy/osu/commit/5d96e6d90a30b85d27866d70ce9ff5f693247e0f, I guess it makes sense for the team editor screen to force ID `1` on invalid.

File: osu.Game.Tournament/Screens/Editors/TeamEditorScreen.cs (hunk 2 above)
Comment (frenzibyte) [reply osu.Game.Tournament/Screens/Editors/TeamEditorScreen.cs]: Right, replaced this with fixing the user ID on a successful request instead.

**Comment by bdach** [comment]
I'm confused by the description. Does this PR fix the last 2 issues listed therein or not? I don't understand the edit.

**Comment by frenzibyte** [comment]
I've corrected it, check again.

**Review by smoogipoo — APPROVED** [review]
i think this looks fine

**Pull request merged**